sycl/source/detail/builtins_relational.cpp

```diff
@@ -21,42 +21,48 @@ __SYCL_INLINE_NAMESPACE(cl) {
 namespace __host_std {
 namespace {
 
-template <typename T> inline T __vFOrdEqual(T x, T y) { return -(x == y); }
+template <typename T> inline T __vFOrdEqual(T x, T y) {
```

Comment:
should these return a bool or T?
Where in the spec, this is stated?

Reply:
`__vFOrdEqual` is not stated in the spec. I am not sure as to if this should return a bool or T.
The goal of this change to resolve an issue with building DPCPP on windows and not to change any functionality.

Reply:
All these are  part of 4.17.9. Relational functions. Some of them should return bool. However, you are right that the PR does not change any functionality. If there is a a problem, it was already there in the original code.

Reply:
BTW, my guess why it returns -1 is because it is often good for vectorizer to have element filled with _ones_, 
but I also was puzzled seeing return of -1.0 for float comparison. 
It would be logically more correct to returning int32_t{-1} for float32 rather than -1.0

Reply:
User-visible methods must do exactly that according to the spec. I guess/think/hope that only those internal functions behave differently for some reason.